weave/flow/obj.py

```diff
@@ -12,6 +13,27 @@
 from weave.trace.vals import WeaveObject, pydantic_getattribute
 from weave.trace.weave_client import get_ref
 
+logger = logging.getLogger(__name__)
+
+T = TypeVar("T")
+
+
+def deprecated_field(new_field_name: str) -> Callable[[Callable[[Any], T]], property]:
```

Comment:
what is this for? not seeing it used anywhere

Reply:
This is being added to facilitate some work @tcapelle is doing where:
1. He wants to change the name of a field; but
2. We want to make it backwards-compatible

https://weightsandbiases.slack.com/archives/C0785BR1RNZ/p1728746308456249?thread_ts=1728633360.048499&cid=C0785BR1RNZ